src/main/java/org/skriptlang/skript/bukkit/potion/elements/effects/EffApplyPotionEffect.java

```diff
@@ -0,0 +1,88 @@
+package org.skriptlang.skript.bukkit.potion.elements.effects;
+
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Example;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.lang.Effect;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.util.Timespan;
+import ch.njol.skript.util.Timespan.TimePeriod;
+import ch.njol.util.Kleenean;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.potion.util.PotionUtils;
+import org.skriptlang.skript.bukkit.potion.util.SkriptPotionEffect;
+import org.bukkit.entity.LivingEntity;
+import org.bukkit.event.Event;
+import org.bukkit.potion.PotionEffect;
+import org.skriptlang.skript.registration.SyntaxInfo;
+import org.skriptlang.skript.registration.SyntaxRegistry;
+
+@Name("Apply Potion Effect")
+@Description("Applies a potion effect to an entity.")
+@Example("apply swiftness 2 to the player")
+@Example("""
+	command /strengthboost:
+		trigger:
+			apply strength 10 to the player for 5 minutes
+""")
+@Example("apply the potion effects of the player's tool to the player")
+// TODO new since
+@Since("2.0, 2.2-dev27 (ambient and particle-less potion effects), 2.5 (replacing existing effect), 2.5.2 (potion effects), INSERT VERSION (syntax changes)")
+public class EffApplyPotionEffect extends Effect {
+
+	public static void register(SyntaxRegistry registry) {
+		// While allowing the user to specify the timespan here is repetitive as you can do it in ExprSecPotionEffect,
+		// it allows syntax like "apply haste 3 to the player for 5 seconds" to work
+		registry.register(SyntaxRegistry.EFFECT, SyntaxInfo.builder(EffApplyPotionEffect.class)
+				.supplier(EffApplyPotionEffect::new)
+				.addPatterns(
+						"apply %skriptpotioneffects% to %livingentities% [for %-timespan%]",
+						"effect %livingentities% with %skriptpotioneffects% [for %-timespan%]"
```

Comment:
I'm never quite sure, should this be affect or effect? I'm pretty sure affect fits better but not 100%. Alternatively, `inflict`?

Reply:
Sounds good